src/tasks/base-chat-task.ts

```diff
@@ -0,0 +1,56 @@
+import { FastifyLoggerInstance } from 'fastify';
+import {
+  Actor,
+  DatabaseTransactionHandler,
+  IndividualResultType,
+  ItemMembershipService,
+  ItemService,
+  Member,
+  PostHookHandlerType,
+  PreHookHandlerType,
+  Task,
+  TaskStatus,
+} from 'graasp';
+import { ChatService } from '../db-service';
+
+export abstract class BaseChatTask<R> implements Task<Actor, R> {
+  protected itemService: ItemService;
+  protected itemMembershipService: ItemMembershipService;
+  protected chatService: ChatService;
+  protected _result: R;
+  protected _message: string;
+
+  readonly actor: Member;
+
+  status: TaskStatus;
+  targetId: string;
+  data: Partial<IndividualResultType<R>>;
+  preHookHandler: PreHookHandlerType<R>;
+  postHookHandler: PostHookHandlerType<R>;
+
+  constructor(
+    actor: Member,
+    itemService: ItemService,
+    itemMembershipService: ItemMembershipService,
+    chatService: ChatService,
+  ) {
+    this.actor = actor;
+    this.status = 'NEW';
```

Comment:
Should we factor this string out? Maybe enum to track status?

Reply:
In Typescript from the perspective of types, unions have the same domain guarantees as enums: `status` is of union type `TaskStatus` defined as follows in the core code:
```ts
export declare type TaskStatus = 'NEW' | 'RUNNING' | 'OK' | 'FAIL' | 'DELEGATED';
```
So in the Typescript boundaries, this type can never be assigned anything else than these values. Arguably enums could help maintainability, but then you need to worry about whether it should be instantiated or inlined (which can cause other issues). Basically union types are zero-cost abstractions as they provide the guarantees at compile-time, and are then inlined into strings for runtime.

I'm using the core code from @ahgentil so currently this is consistent with the rest of the server and plugins.